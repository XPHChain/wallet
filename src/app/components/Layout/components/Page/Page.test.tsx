import { Contracts } from "@payvo/profiles";
import electron from "electron";
import { createMemoryHistory } from "history";
import React from "react";
import { Route } from "react-router-dom";
import { act, env, fireEvent, getDefaultProfileId, renderWithRouter } from "utils/testing-library";

import { Page } from "./Page";

let profile: Contracts.IProfile;

const dashboardURL = `/profiles/${getDefaultProfileId()}/dashboard`;
const history = createMemoryHistory();

describe("Page", () => {
	beforeAll(() => {
		profile = env.profiles().findById(getDefaultProfileId());
		history.push(dashboardURL);
	});

	it("should render", () => {
		const sidebar = true;

		const { container, asFragment } = renderWithRouter(
			<Page title="Test" sidebar={sidebar}>
				{}
			</Page>,
		);

		expect(container).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render without sidebar", () => {
		const { container, asFragment } = renderWithRouter(<Page title="Test">{}</Page>);

		expect(container).toBeInTheDocument();
		expect(asFragment()).toMatchSnapshot();
	});

	it.each(["Contacts", "Votes", "Settings", "Support"])(
		"should handle '%s' click on user actions dropdown",
		async (label) => {
			const ipcRendererSpy = jest.spyOn(electron.ipcRenderer, "send").mockImplementation();
			const historySpy = jest.spyOn(history, "push").mockImplementation();

			const { getByTestId, findByText, findByTestId } = renderWithRouter(
				<Route path="/profiles/:profileId/dashboard">
					<Page>{}</Page>
				</Route>,
				{
					history,
					routes: [dashboardURL],
				},
			);

			await findByTestId("navbar__useractions");

			const toggle = getByTestId("navbar__useractions");

			act(() => {
				fireEvent.click(toggle);
			});

			await findByText(label);

			fireEvent.click(await findByText(label));

			if (label === "Support") {
				expect(ipcRendererSpy).toHaveBeenCalledWith("open-external", "https://payvo.com/contact");
			} else {
				expect(historySpy).toHaveBeenCalledWith(`/profiles/${profile.id()}/${label.toLowerCase()}`);
			}

			ipcRendererSpy.mockRestore();
			historySpy.mockRestore();
		},
	);

	it("should handle 'Sign Out' click on user actions dropdown", async () => {
		const { getByTestId, findByText, findByTestId } = renderWithRouter(
			<Route path="/profiles/:profileId/dashboard">
				<Page>{}</Page>
			</Route>,
			{
				history,
				routes: [dashboardURL],
			},
		);

		const historySpy = jest.spyOn(history, "push").mockImplementation();

		await findByTestId("navbar__useractions");

		const toggle = getByTestId("navbar__useractions");

		act(() => {
			fireEvent.click(toggle);
		});

		await findByText("Sign Out");

		fireEvent.click(await findByText("Sign Out"));

		expect(historySpy).toHaveBeenCalledWith("/");

		historySpy.mockRestore();
	});
});
