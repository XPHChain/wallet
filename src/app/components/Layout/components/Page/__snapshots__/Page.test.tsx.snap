// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page should render 1`] = `
<DocumentFragment>
  <div
    class="flex relative flex-col min-h-screen"
  >
    <nav
      aria-labelledby="main menu"
      class="NavigationBar__NavWrapper-sc-1nkmx4c-0 nFzZM"
    >
      <div
        class="flex relative h-21"
      >
        <button
          class="BackButton__StyledBackButton-sc-18ta3r7-0 tELEj flex w-12"
        >
          <div
            class="sc-bdnxRM AkeAK mx-auto"
            height="10"
            width="10"
          >
            <svg>
              chevron-left-small.svg
            </svg>
          </div>
        </button>
        <div
          class="flex flex-1 px-8"
        >
          <div
            class="flex items-center my-auto"
          >
            <div
              class="NavigationBar__LogoContainer-sc-1nkmx4c-1 bfIa-di"
            >
              <svg
                height="28"
              >
                payvo-logo.svg
              </svg>
            </div>
            <span
              class="text-2xl font-bold"
            >
              Test
            </span>
          </div>
          <ul
            class="flex mr-auto ml-4 space-x-8 h-21"
          />
          <div
            class="flex items-center my-auto space-x-4"
          >
            <div
              class="h-8 border-r border-theme-secondary-300 dark:border-theme-secondary-800"
            />
            <div
              class="flex items-center"
            >
              <div
                class="NavigationBar__NavigationButtonWrapper-sc-1nkmx4c-2 jTnlvd"
              >
                <button
                  class="OriginalButton__StyledButton-sc-o0fs2j-0 dMZjng"
                  data-testid="navbar__buttons--send"
                  disabled=""
                  type="button"
                >
                  <div
                    class="flex items-center space-x-2"
                  >
                    <div
                      class="sc-bdnxRM NjstW p-1"
                      height="20"
                      width="20"
                    >
                      <svg>
                        double-arrow-right.svg
                      </svg>
                    </div>
                  </div>
                </button>
              </div>
            </div>
            <div
              class="h-8 border-r border-theme-secondary-300 dark:border-theme-secondary-800"
            />
            <div
              class="flex items-center"
            >
              <div
                class="NavigationBar__NavigationButtonWrapper-sc-1nkmx4c-2 jTnlvd"
              >
                <button
                  class="OriginalButton__StyledButton-sc-o0fs2j-0 dMZjng"
                  data-testid="navbar__buttons--receive"
                  disabled=""
                  type="button"
                >
                  <div
                    class="flex items-center space-x-2"
                  >
                    <div
                      class="sc-bdnxRM NjstW p-1"
                      height="20"
                      width="20"
                    >
                      <svg>
                        qr-code.svg
                      </svg>
                    </div>
                  </div>
                </button>
              </div>
            </div>
            <div
              class="h-8 border-r border-theme-secondary-300 dark:border-theme-secondary-800"
            />
          </div>
          <div
            class="flex items-center my-auto ml-8 space-x-4"
          >
            <div
              class="text-right"
            >
              <div
                class="text-xs font-semibold text-theme-secondary-500"
              >
                Your Balance
              </div>
              <div
                class="text-sm font-bold text-theme-secondary-text dark:text-theme-text"
                data-testid="Balance__value"
              >
                <span
                  class="whitespace-nowrap"
                  data-testid="AmountCrypto"
                >
                  0 BTC
                </span>
              </div>
            </div>
            <div
              class="relative"
            >
              <span
                data-testid="dropdown__toggle"
              >
                <div
                  class="relative justify-center items-center align-middle rounded-full cursor-pointer"
                  data-testid="navbar__useractions"
                >
                  <div
                    class="Avatar__AvatarWrapper-sc-18vw8a4-0 kkxZqy flex-shrink-0"
                    data-testid="Avatar"
                  >
                    <div
                      class="w-full h-full inline-flex items-center justify-center overflow-hidden align-middle rounded-full"
                    >
                      <img
                        alt="Profile Avatar"
                        class="object-cover w-11 h-11 bg-center bg-no-repeat bg-cover rounded-full"
                      />
                    </div>
                  </div>
                </div>
              </span>
            </div>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="flex flex-col flex-1"
    >
      <div
        class="flex flex-1"
      >
        <div
          class="container flex mx-auto"
        >
          <div
            class="px-12 my-8 border-r border-theme-primary-100 dark:border-theme-secondary-800"
          />
          <div
            class="w-full"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Page should render without sidebar 1`] = `
<DocumentFragment>
  <div
    class="flex relative flex-col min-h-screen"
  >
    <nav
      aria-labelledby="main menu"
      class="NavigationBar__NavWrapper-sc-1nkmx4c-0 nFzZM"
    >
      <div
        class="flex relative h-21"
      >
        <button
          class="BackButton__StyledBackButton-sc-18ta3r7-0 tELEj flex w-12"
        >
          <div
            class="sc-bdnxRM AkeAK mx-auto"
            height="10"
            width="10"
          >
            <svg>
              chevron-left-small.svg
            </svg>
          </div>
        </button>
        <div
          class="flex flex-1 px-8"
        >
          <div
            class="flex items-center my-auto"
          >
            <div
              class="NavigationBar__LogoContainer-sc-1nkmx4c-1 bfIa-di"
            >
              <svg
                height="28"
              >
                payvo-logo.svg
              </svg>
            </div>
            <span
              class="text-2xl font-bold"
            >
              Test
            </span>
          </div>
          <ul
            class="flex mr-auto ml-4 space-x-8 h-21"
          />
          <div
            class="flex items-center my-auto space-x-4"
          >
            <div
              class="h-8 border-r border-theme-secondary-300 dark:border-theme-secondary-800"
            />
            <div
              class="flex items-center"
            >
              <div
                class="NavigationBar__NavigationButtonWrapper-sc-1nkmx4c-2 jTnlvd"
              >
                <button
                  class="OriginalButton__StyledButton-sc-o0fs2j-0 dMZjng"
                  data-testid="navbar__buttons--send"
                  disabled=""
                  type="button"
                >
                  <div
                    class="flex items-center space-x-2"
                  >
                    <div
                      class="sc-bdnxRM NjstW p-1"
                      height="20"
                      width="20"
                    >
                      <svg>
                        double-arrow-right.svg
                      </svg>
                    </div>
                  </div>
                </button>
              </div>
            </div>
            <div
              class="h-8 border-r border-theme-secondary-300 dark:border-theme-secondary-800"
            />
            <div
              class="flex items-center"
            >
              <div
                class="NavigationBar__NavigationButtonWrapper-sc-1nkmx4c-2 jTnlvd"
              >
                <button
                  class="OriginalButton__StyledButton-sc-o0fs2j-0 dMZjng"
                  data-testid="navbar__buttons--receive"
                  disabled=""
                  type="button"
                >
                  <div
                    class="flex items-center space-x-2"
                  >
                    <div
                      class="sc-bdnxRM NjstW p-1"
                      height="20"
                      width="20"
                    >
                      <svg>
                        qr-code.svg
                      </svg>
                    </div>
                  </div>
                </button>
              </div>
            </div>
            <div
              class="h-8 border-r border-theme-secondary-300 dark:border-theme-secondary-800"
            />
          </div>
          <div
            class="flex items-center my-auto ml-8 space-x-4"
          >
            <div
              class="text-right"
            >
              <div
                class="text-xs font-semibold text-theme-secondary-500"
              >
                Your Balance
              </div>
              <div
                class="text-sm font-bold text-theme-secondary-text dark:text-theme-text"
                data-testid="Balance__value"
              >
                <span
                  class="whitespace-nowrap"
                  data-testid="AmountCrypto"
                >
                  0 BTC
                </span>
              </div>
            </div>
            <div
              class="relative"
            >
              <span
                data-testid="dropdown__toggle"
              >
                <div
                  class="relative justify-center items-center align-middle rounded-full cursor-pointer"
                  data-testid="navbar__useractions"
                >
                  <div
                    class="Avatar__AvatarWrapper-sc-18vw8a4-0 kkxZqy flex-shrink-0"
                    data-testid="Avatar"
                  >
                    <div
                      class="w-full h-full inline-flex items-center justify-center overflow-hidden align-middle rounded-full"
                    >
                      <img
                        alt="Profile Avatar"
                        class="object-cover w-11 h-11 bg-center bg-no-repeat bg-cover rounded-full"
                      />
                    </div>
                  </div>
                </div>
              </span>
            </div>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="flex flex-col flex-1"
    />
  </div>
</DocumentFragment>
`;
